### ISSUE TITLE
RX - Update Design for List and Details View 

### ISSUE THREAD

**Issue by bischoffa** [department-of-veterans-affairs/va-mobile-app#7355]
## Description 
As a VA, I want to update the RX Sort/Filter function, so that it may benefit the Veteran users more. 

As part of this story we want to:   

- Address screen component IA
    - Remove tabs
    - Reorder some screen components 
- Revise the current [Filter by / Sort by] experience:
    - Introduce dynamic headers that follow the selected Filter
    - Add a panel that Veteran can filter and sort by in a single place
    - Retain [Cancel] (top left corner)
    - Replace [Reset] with [Apply] (top right corner)
    - Remove the [Apply] footer.button 

---
#### 1. Introduce dynamic list titles
 - [ ] Dynamic list title
     - When All/Active/Discontinued/Expired/Transferred/Unknown filters are selected, the header will be: [Filter option] prescriptions (# of RX), sorted by [sort option]
     - When Pending filter is selected, the header will be: Pending refills (x of RX), sorted by [sort option]
     - When Tracking filter is selected, the header will be: Refills with tracking (x of RX), sorted by [sort option]
     - Sort option isn't capitalized in this header (i.e., ...sorted by status (A to Z), NOT ...sorted by Status (A to Z)
     - [ ] Dynamic description under list title
     - When All/Active/Discontinued/Expired/Transferred/Unknown filters are selected, the description under the header will be: "This list only shows prescriptions filled by VA pharmacies and may not include all your medications." 
     - When Pending filter is selected, the description will be: "This list shows refill requests you've submitted. It also shows refills the VA pharmacy is processing."
     - When Tracking filter is selected, the description will be: "This list shows refills with current tracking information available."
 ---
#### 2. Revise current experience
- [ ] Remove current [Filter by / Sort by] buttons
-  [ ] Remove [Sort by] panel 
-  [ ] Create "Filter and sort" Secondary Button 
-  [ ] Create a panel with a "Filter and sort" header 
    - [ ] Add Cancel - top left
    - [ ] Add Apply - top right
---
#### 3. Filter + Sort Panel
 - [ ] Create "Filter by" section   
    -  [ ] All (with number of RX with this status) 
    -  [ ] Active (with number of RX with this status) 
       -  [ ] Apply Helper Text style: "Includes these statuses: On hold, Parked, Refill in process, and Submitted"
    -  [ ] Discontinued (with number of RX with this status)
    -  [ ] Expired (with number of RX with this status)
    -  [ ] Pending (with number of RX that fit this) 
       -  [ ] Apply Helper Text style: "Includes refill requests you submitted and refills the VA pharmacy is processing"
    -  [ ] Tracking (with number of RX that fit this) 
       -  [ ] Apply Helper Text style: "Includes refills with current tracking information available"
    -  [ ] Transferred (with number of RX with this status) 
    -  [ ] Unknown (with number of RX with this status)
-  [ ] Create "Sort by" section 
    -  [ ] Fill date (newest to oldest)
    -  [ ] Medication name (A to Z)
    -  [ ] Refills left (least to most)
    -  [ ] Status (A to Z) 
- [ ] Enable Default states 
    - Filter by = All 
    - Sort by = Status (A to Z)

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNWJ3QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--3be27d3f40684a1aaa357008669a4096f3c688fd/image.png)
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNWZ3QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--df694cafb522dcb51acfea932aade6ac70d16654/image.png)
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNWp3QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--708ed112562a0fffe008a3b3fa00a10075aff5c1/image.png)

**Accessibility Requirements**
- [ ] Screen reader accessible 

## Notes & Open Questions
- [Figma prototype ](https://www.figma.com/proto/Si4LF4t8VOv8emdCovv3U3/%F0%9F%94%8D-Prescriptions-2.0---Working---VAMobile?page-id=1499%3A15245&type=design&node-id=1499-15407&viewport=812%2C486%2C0.31&t=gjAhJIZN0xMTxh98-1&scaling=min-zoom&starting-point-node-id=1499%3A15407)
- [Figma Prototype instructions](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/va-mobile-app/features/medication/Design/List%20enhancements%20Q4%202023.md)
- Dr Spahn feedback
    - Have the Filter option be multi-select (such as a user can select Active and Pending) - Kay / Coulton agreed to move forward without this 
       -  Rationale: (making filter options multi-select), that'd require more content changes to the header on the landing screen. That header is dynamic based on the filter option selected. So a multi-select could make that header unhelpful (and too long).
    - Bi-directional sorting- A to Z and Z to A).  - Kay / Coulton agreed to move forward without this 
       -  Rationale: he reason we didn't add bi-directional sort (we didn't add both options (A-Z and Z-A)) because if we did that for every sort option. it'd make the screen (large panel) very long...especially since we're now including filter options with sort options
- Only able to select 1 filter and 1 sort by 
- Default filter and sort is - Filter = All; Sort by = Status A to Z 
- Dependencies/Roadblocks: Nope Dr. Spahn approved as of 11/20/23
- Any internal/external dependencies? Yes - test accounts with the necessary statuses to test 
- Test accounts needed? Yes
- Does this require QA? Yes
    
- Dev Notes: 

## Ticket Checklist

- [X] Acceptance criteria defined
- [X]  Labels added (front-end, back-end, feature)
- [X] Linked to an Epic

**Comment by Sparowhawk** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Mobile-60f1a34998bc75000f2a489f/issues/department-of-veterans-affairs/va-mobile-app/7355?planning-poker) with Zenhub @alexandec

**Comment by Sparowhawk** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Mobile-60f1a34998bc75000f2a489f/issues/department-of-veterans-affairs/va-mobile-app/7355?planning-poker) with Zenhub @Sparowhawk

**Comment by alexandec** [comment]
@kidlaurenpeace how should this screen look when there are no matches? Something like this?

<img src="https://github.com/department-of-veterans-affairs/va-mobile-app/assets/5805150/fc20f9cd-6120-401a-ac32-16f9ebf16bc8" width="49%" />

**Comment by kidlaurenpeace** [comment]
@alexandec Yes that's fine.

**Comment by DJUltraTom** [comment]
Noting on here that this will require rewriting QA's manual test suite for RX as well as requiring updated Detox automation.

**Comment by alexandec** [comment]
Regarding the E2E (detox) tests, I updated `e2e/tests/Prescriptions.e2e.ts` to reflect these changes. That's included in the PR for this ticket. I tagged @rbontrager in the PR so she's aware and can take a look when she gets a chance.

**Comment by timwright12** [comment]
I moved this back into code review so the e2e test could be reviewed before QA

**Comment by rbontrager** [comment]
The E2E tests were reviewed and approved in [7515](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/7515).

**Comment by rbontrager** [comment]
RX changes look good.  Slightly changed some of our RX testRail cases here: [RX revision](https://dsvavsp.testrail.io/index.php?/suites/view/92&group_by=cases:section_id&group_id=13043&group_order=asc&display_deleted_cases=0). @DJUltraTom When you get back can you do a quick check that they look good so QA can approve this ticket.

**Comment by DJUltraTom** [comment]
RX revision test cases look good, ran through fine, QA approves

**Comment by DJUltraTom** [comment]
waiting on all items in [7487](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/7487) to be completed before final validation and sign off can occur

**Comment by bischoffa** [comment]
@DJUltraTom looks like you are good to go from chatting with Dylan and Chris. 
The call center guide was submitted. App store images need approved by Alayna. Neither though should block final validation. 
Asked Chris / Dylan to bring this up in H&B standup.

**Comment by bischoffa** [comment]
Per convo last week RX will not be in the sprint release. It will go into the following release 2.23

**Comment by alexandec** [comment]
@bischoffa I removed the 2.22.0 release tag since this isn't going in that release

**Comment by alexandec** [comment]
@DJUltraTom all the RX changes are now merged into develop and ready for final testing in that branch

**Comment by DJUltraTom** [comment]
verified this is looking good on both iOS and android on develop.
Closing as "Deployed"
